x-pack/packages/kbn-elastic-assistant/impl/connectorland/connector_selector_inline/connector_selector_inline.tsx

```diff
@@ -0,0 +1,286 @@
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiButtonEmpty, EuiFlexGroup, EuiFlexItem, EuiSuperSelect, EuiText } from '@elastic/eui';
+import React, { useCallback, useMemo, useState } from 'react';
+
+import { ActionConnector } from '@kbn/triggers-actions-ui-plugin/public';
+
+import { ActionConnectorProps } from '@kbn/triggers-actions-ui-plugin/public/types';
+import { ConnectorAddModal } from '@kbn/triggers-actions-ui-plugin/public/common/constants';
+import {
+  GEN_AI_CONNECTOR_ID,
+  OpenAiProviderType,
+} from '@kbn/stack-connectors-plugin/public/common';
+import { css } from '@emotion/css/dist/emotion-css.cjs';
+import { Conversation } from '../../..';
+import { useLoadConnectors } from '../use_load_connectors';
+import * as i18n from '../translations';
+import { useLoadActionTypes } from '../use_load_action_types';
+import { useAssistantContext } from '../../assistant_context';
+import { useConversation } from '../../assistant/use_conversation';
+
+export const ADD_NEW_CONNECTOR = 'ADD_NEW_CONNECTOR';
+interface Props {
+  isDisabled?: boolean;
+  onConnectorSelectionChange: (connectorId: string, provider: OpenAiProviderType) => void;
+  selectedConnectorId?: string;
+  selectedConversation?: Conversation;
+  onConnectorModalVisibilityChange?: (isVisible: boolean) => void;
+}
+
+interface Config {
+  apiProvider: string;
+}
+
+const inputContainerClassName = css`
+  height: 32px;
+
+  .euiSuperSelect {
+    width: 400px;
+  }
+
+  .euiSuperSelectControl {
+    border: none;
+    box-shadow: none;
+    background: none;
+    padding-left: 0;
+  }
+
+  .euiFormControlLayoutIcons {
+    right: 14px;
+    top: 2px;
+  }
+`;
+
+const inputDisplayClassName = css`
+  overflow: hidden;
+  text-overflow: ellipsis;
+  max-width: 400px;
+`;
+
+const placeholderButtonClassName = css`
+  overflow: hidden;
+  text-overflow: ellipsis;
+  max-width: 400px;
+  font-weight: normal;
+  padding-bottom: 5px;
+  padding-left: 0;
+  padding-top: 2px;
+`;
+
+/**
+ * A minimal and connected version of the ConnectorSelector component used in the Settings modal.
+ */
+export const ConnectorSelectorInline: React.FC<Props> = React.memo(
+  ({
+    isDisabled = false,
+    onConnectorModalVisibilityChange,
+    selectedConnectorId,
+    selectedConversation,
+    onConnectorSelectionChange,
+  }) => {
+    const [isOpen, setIsOpen] = useState<boolean>(false);
+    const { actionTypeRegistry, http } = useAssistantContext();
+    const { setApiConfig } = useConversation();
+    // Connector Modal State
+    const [isConnectorModalVisible, setIsConnectorModalVisible] = useState<boolean>(false);
+    const { data: actionTypes } = useLoadActionTypes({ http });
+    const actionType = actionTypes?.find((at) => at.id === GEN_AI_CONNECTOR_ID) ?? {
+      enabledInConfig: true,
+      enabledInLicense: true,
+      minimumLicenseRequired: 'platinum',
```

Comment:
nit: should this be `enterprise`?

Reply:
I believe this is just to satisfy the types in the case of a fallback if there is no queried `GEN_AI_CONNECTOR_ID`. In that case it should use the fallback's `id: '.gen-ai',` (and not license) to determine what type of connector add modal to show.